### DISCUSSION TITLE
"Unable to render this definition" error solutions not working

### DISCUSSION THREAD

**Discussion by celinehaddad** [swagger-api/swagger-ui#8586] [Q&A]
Hey swagger team, 

I need some help with implementing swagger-ui-react in my Next.js app. I found this issue thread https://github.com/swagger-api/swagger-ui/issues/8245 to solve my initial error message ("SyntaxError: Cannot use import statement outside a module") but I tried [this solution](https://github.com/swagger-api/swagger-ui/issues/8245#issuecomment-1433849504) and I get this error: 

![Screen Shot 2023-04-21 at 10 53 34 PM](https://user-images.githubusercontent.com/68402310/233758204-54c070cb-d286-4678-81ad-b41c8606c577.png)

I also noticed that when I change the spec file (was testing json and yml formats if it would behave differently- it doesn't), the spec file didn't change from json to yml based on what I specified..it just kept the first file I tried & I verified this in my console.log. SO, I tried a different solution I found in that issue thread to dynamically import swagger-ui-react, and that solved the "spec file not changing  to what I define" issue. But, I still get the same exact error message as pictured above. 

My spec file specifies  "openapi": "3.0.2" at the top. I tested changing it to "3.0.1" & tested adding "swagger": "2.0" above the openapi version as well, but none of this worked & I still get that error message. I tested this out at https://editor.swagger.io/ & the UI of my openapi spec file renders perfectly. 

Not sure what I need to do to get this working as my openapi spec meets the requirements of the error message. Any ideas?

**Comment by Di3Z1E** [comment]
Have you managed yo overcome this error?
I'm facing a similar one, and yet cannot find a solution to it

**Reply by ruchidavda1** [reply]
same issue

**Reply by Ashwith-Madikonda** [reply]
same here

**Reply by joaofrca** [reply]
same here!

**Reply by kj-brown** [reply]
As above

**Reply by smdurjoy** [reply]
Same here!!

**Reply by cshilton** [reply]
Also here.
